src/main/java/Deadline.java

```diff
@@ -0,0 +1,35 @@
+public class Deadline extends Task{
+
+    // Stores information about deadline of task
+    private String by;
+
+    // Overloaded Constructor
+    public Deadline(String descriptionWithDeadline){
+        super(getDescription(descriptionWithDeadline));
+        taskType = 'D';
+        this.by = getDeadline(descriptionWithDeadline);
+    }
+
+    // Abstracts out the description from the user given input and returns it
+    private static String getDescription(String descriptionWithDeadline) {
+        String[] split = descriptionWithDeadline.split("/",2);
+        return split[0];
+    }
+
+    // Abstracts out the deadline from the user given input and returns it
+    private static String getDeadline(String descriptionWithDeadline) {
+        String[] split = descriptionWithDeadline.split("/",2);
```

Comment:
Can improve variable name "split" eg. words?

Reply:
Thanks for the suggestion